packages/umi/src/PublicKey.ts

```diff
@@ -7,158 +9,212 @@ import { base58, uniqueBy } from './utils';
 export const PUBLIC_KEY_LENGTH = 32;
 
 /**
- * Defines an object that has a public key.
+ * Defines a public key as a base58 string.
  * @category Signers and PublicKeys
  */
-export type HasPublicKey = { publicKey: PublicKey };
+export type PublicKey = Base58EncodedAddress;
```

Comment:
Things get _really_ interesting when you make this parametric on the actual key.

```ts
export type PublicKey<TAddress extends string> = Base58EncodedAddress<TAddress>;
```

Highly recommend.

Example of why this is useful: https://github.com/solana-labs/solana-web3.js/pull/1347